### PR TITLE
Adds content-type message header to support other content-types such HTML

### LINKED ISSUE #112: Accept message header field: content-type to support HTML & plain text
**Is your feature request related to a problem? Please describe.**
I want to send html-based messages so I can format messages and include pictures 

**Describe the solution you'd like**
The application may use a `Content-Type: multipart/alternative` be able to include message parts:
* `text/html`
* `text/plain`

**Describe alternatives you've considered**
Sending the message with ascii art or attaching pictures to the message. Neither work particularly well.

**Additional context**
See https://tools.ietf.org/html/rfc1521#section-7.2 for multipart detail

### LINKED ISSUE #250: Include Content-Type and Content-Transfer-Encoding
**Is your feature request related to a problem? Please describe.**
API consumers can not tell what the content type is when reading a message, this prevents the UI from knowing if its HTML or plaintext messages. This also allows for displaying UTF-8 and UTF-16 character sets correctly.

**Describe the solution you'd like**
Extend http://127.0.0.1:8080/api/docs#operation/GetMessages response json `headers` to include `content-type` and `content-type-encoding`.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by nikosk686**
Closes #112
Closes #250

I use `http.DetectContentType(data []byte)` method to 
automatically detect the type from the message's body.
This seems to work pretty good, especially for simple cases as HTML.
I prefer this solution as it does not need any extra info from the client.
However the implementation can be tuned, to use a value provided by the client.

Also, this patch does not perform any changes regarding `Content-Transfer-Encoding`.

### CODE REVIEW

**Comment by nikosk686** [comment]
The build seems to have failing tests..However locally all tests pass..I will check again

**Comment by robdefeo** [comment]
Try running `make unit-test` and it will do a full test run of all packages.

**Comment by nikosk686** [comment]
@robdefeo, still no errors locally..

**Comment by robdefeo** [comment]
I can't make a change to your branch on my machine so had to do a couple of changes through the UI. The problem was that they lines were being terminated by a LF rather than an CRLF. There is a mode you can change in your editor, or you can copy the simple file and change everything except the final line ending. Take a look at the changes and build and tell me if you have any questions.

**Comment by robdefeo** [comment]
A quick note on why this is important, Mailchain uses a SHA3-256 hash of the message bytes to generate a secure hash which is stored on the blockchain. Even if the file looks visually the same a change in a single byte causes a completely different hash result. This is a a characteristic of the hash function make it more secure.

**Comment by nikosk686** [comment]
It seems that git changes crlf to lf automatically on my machine (linux).
I am trying to configure it.

**Comment by nikosk686** [comment]
Ok fixed! I have added a .gitattributes file with this `*eml text eol=crlf`.
That way we ensure that this won't happen again on .eml files regardless 
which OS.

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1 @@
+*eml text eol=crlf
```
Comment (robdefeo) [inline .gitattributes]: Great idea 👍

**Pull request merged**